mentoring-app/src/app/app.component.spec.ts

```diff
@@ -1,11 +1,28 @@
 import { TestBed, async } from '@angular/core/testing';
 import { AppComponent } from './app.component';
+import { Component } from '@angular/core';
+
+@Component({selector: 'app-header', template: ''})
+class HeaderStubComponent {}
+
+@Component({selector: 'router-outlet', template: ''})
+class RouterOutletStubComponent { }
+
+@Component({selector: 'app-breadcrumbs', template: ''})
+class BreadcrumbStubComponent {}
+
+@Component({selector: 'app-footer', template: ''})
+class FooterStubComponent {}
 
 describe('AppComponent', () => {
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       declarations: [
-        AppComponent
+        AppComponent,
+        HeaderStubComponent,
+        RouterOutletStubComponent,
+        BreadcrumbStubComponent,
+        FooterStubComponent
```

Comment:
You could have used schemas option of @NgModule instead:
schemas: [CUSTOM_ELEMENTS_SCHEMA]

Reply:
fixed for app.component, but left for courses page, just to try dirrerent approaches